.github/workflows/kfp-backend-postgres-tests.yml

```diff
@@ -0,0 +1,245 @@
+          ref: ${{ github.event.inputs.test_ref }}
+          fetch-depth: 0
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.22"
+          cache: true
+
+      - name: Create KinD cluster
+        uses: helm/kind-action@v1
+        with:
+          cluster_name: kfp-ci
+          wait: 300s
+
+      - name: Install kubectl
+        uses: azure/setup-kubectl@v4
+        with:
+          version: "latest"
+
+      - name: Install kustomize
+        uses: imranismail/setup-kustomize@v2
+        with:
+          kustomize-version: "5.4.2"
+
+      - name: Install utils (psql, nc, jq)
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y postgresql-client netcat-openbsd jq
+
+      - name: Create namespace
+        env:
+          NAMESPACE: kfp-pgx-test
+        run: kubectl create namespace "$NAMESPACE" || true
+
+      - name: Install default StorageClass for KinD (local-path)
+        run: |
+          set -euo pipefail
+
+          # 1) Install local-path-provisioner (idempotent)
+          kubectl apply -f https://raw.githubusercontent.com/rancher/local-path-provisioner/master/deploy/local-path-storage.yaml
+
+          # 2) Detect its namespace (usually local-path-storage)
+          LPP_NS=$(kubectl get deploy -A -o jsonpath='{range .items[?(@.metadata.name=="local-path-provisioner")]}{.metadata.namespace}{"\n"}{end}')
+          if [ -z "${LPP_NS:-}" ]; then
+            echo "Failed to locate local-path-provisioner deployment"
+            kubectl get deploy -A
+            exit 1
+          fi
+          echo "local-path-provisioner namespace: $LPP_NS"
+
+          # 3) Wait for controller to be ready
+          kubectl -n "$LPP_NS" rollout status deploy/local-path-provisioner --timeout=180s
+
+          # 4) Mark StorageClass as default (idempotent)
+          if ! kubectl get storageclass local-path >/dev/null 2>&1; then
+            echo "StorageClass 'local-path' not found, listing SCs:"
+            kubectl get storageclass -o wide
+            exit 1
+          fi
+          kubectl annotate storageclass local-path storageclass.kubernetes.io/is-default-class="true" --overwrite
+
+          echo "StorageClasses:"
+          kubectl get storageclass
+
+          # 5) Verify default StorageClass exists
+          kubectl get storageclass | tee /dev/stderr | grep "(default)"
+
+      - name: Wait for default StorageClass present
+        run: |
+          # Sanity check: ensure there is a default StorageClass before deploying PG
+          kubectl get storageclass | tee /dev/stderr | grep "(default)" >/dev/null
+
+      - name: Deploy Postgres via Kustomize
+        working-directory: manifests/kustomize/third-party/postgresql
+        env:
+          NAMESPACE: kfp-pgx-test
+        run: |
+          kustomize build ./base | kubectl -n "$NAMESPACE" apply -f -
+          kubectl -n "$NAMESPACE" get all
```

Comment:
can we instead update the `deploy-kfp.sh` to take [another flag](https://github.com/kubeflow/pipelines/blob/c11a7358f566be93c65131cf3f3cfbc722b40080/.github/resources/scripts/deploy-kfp.sh#L35) and deploy postgresql option? then use the [kfp deploy cluster action](https://github.com/kubeflow/pipelines/blob/9526abb03d603b0d367fd2f13c8a1949388621a5/.github/workflows/e2e-test.yml#L42). 

This script will also wait for pods to ready so you can skip the `Wait for Postgres pod ready` step as well.